Engine/src/Engine/Scene.h

```diff
@@ -3,6 +3,13 @@
 #include "entt.h"
 #include "Components.h"
 
+//Added for UI
+#include <array>
+#include "ExplorerPanel.h"
+#include "EntitiesPanel.h"
+#include "ComponentsPanel.h"
+#include "MainMenu.h"
+
```

Comment:
I think these can be forward declared so that the headers are only included in the cpp. 
If you can't though, then feel free to disregard this.

Reply:
These are all non pointer members so they can't be forward declared